mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -0,0 +1,212 @@
+        scores : pd.Series
+            Scores from running the configurations. The index is the same as the index of the configurations.
+
+        context : pd.DataFrame
+            Not Yet Implemented.
+        """
+        from smac.runhistory import StatusType, TrialInfo, TrialValue  # pylint: disable=import-outside-toplevel
+
+        if context is not None:
+            raise NotImplementedError()
+
+        # Register each trial (one-by-one)
+        for config, score in zip(self._to_configspace_configs(configurations), scores.tolist()):
+            # Retrieve previously generated TrialInfo (returned by .ask()) or create new TrialInfo instance
+            info: TrialInfo = self.trial_info_map.get(config, TrialInfo(config=config))
+            value: TrialValue = TrialValue(cost=score, time=0.0, status=StatusType.SUCCESS)
+            self.base_optimizer.tell(info, value, save=False)
+
+        # Save optimizer once we register all configs
+        self.base_optimizer.optimizer.save()
+
+    def _suggest(self, context: Optional[pd.DataFrame] = None) -> pd.DataFrame:
+        """Suggests a new configuration.
+
+        Parameters
+        ----------
+        context : pd.DataFrame
+            Not Yet Implemented.
+
+        Returns
+        -------
+        configuration : pd.DataFrame
+            Pandas dataframe with a single row. Column names are the parameter names.
+        """
+        from smac.runhistory import TrialInfo  # pylint: disable=import-outside-toplevel
+
+        if context is not None:
+            raise NotImplementedError()
+
+        trial: TrialInfo = self.base_optimizer.ask()
+        self.trial_info_map[trial.config] = trial
+        return pd.DataFrame([trial.config], columns=self.optimizer_parameter_space.get_hyperparameter_names())
+
+    def register_pending(self, configurations: pd.DataFrame, context: Optional[pd.DataFrame] = None) -> None:
+        raise NotImplementedError()
+
+    def surrogate_predict(self, configurations: pd.DataFrame, context: Optional[pd.DataFrame] = None) -> npt.NDArray:
+        from smac.utils.configspace import convert_configurations_to_array  # pylint: disable=import-outside-toplevel
+
+        if context is not None:
+            raise NotImplementedError()
+        if self._space_adapter:
+            raise NotImplementedError()
+
+        if len(self._observations) < self.base_optimizer._initial_design._n_configs:  # pylint: disable=protected-access
+            raise RuntimeError('Surrogate model can make predictions *only* after all initial points have been evaluated')
+
+        configs: npt.NDArray = convert_configurations_to_array(self._to_configspace_configs(configurations))
+        mean_predictions, _ = self.base_optimizer._config_selector._model.predict(configs)  # pylint: disable=protected-access
+        return mean_predictions.reshape(-1,)
+
+    def acquisition_function(self, configurations: pd.DataFrame, context: Optional[pd.DataFrame] = None) -> npt.NDArray:
```

Comment:
Reposting comment from prior rendition that seems to have gotten lost:

```suggestion
    def invoke_acquisition_function(self, configurations: pd.DataFrame, context: Optional[pd.DataFrame] = None) -> npt.NDArray:
```

or

```suggestion
    def aquire(self, configurations: pd.DataFrame, context: Optional[pd.DataFrame] = None) -> npt.NDArray:
```

Thoughts?  @motus

Reply:
We could also split this out to a separate PR.

Reply:
I like renaming, but this definitely should be a separate PR. Also, `s/aquire/acquire/`